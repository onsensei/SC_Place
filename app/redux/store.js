import createSagaMiddleware from 'redux-saga';
import rootReducer from './reducers/index.reducer';
import rootSaga from './sagas/index.saga';
import {applyMiddleware, compose, createStore} from 'redux';
import {fork} from 'redux-saga/effects';
// import * as actions from './actions/index.action';

// middleware

const composeEnhancers = global.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ ? global.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({}) : compose;

const sagaMiddleware = createSagaMiddleware();

const greetingMiddleware = () => (next) => (action) => {
  // if (action.type === actions.GOOD_NIGHT_ACTION) {
  //   action.payload = 'Sweet Dream';
  // }

  next(action);
};

const composedEnhancer = composeEnhancers(applyMiddleware(sagaMiddleware, greetingMiddleware));

function* sagas () {
  yield fork(rootSaga);
}

// store

export const initStore = () => {
  const store = createStore(rootReducer, {}, composedEnhancer);
  sagaMiddleware.run(sagas);
  return store;
};
