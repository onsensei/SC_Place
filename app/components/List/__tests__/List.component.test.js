import ConnectedList, {mapDispatchToProps} from '../List.component';

// Note: test renderer must be required after react-native.
import React from 'react';
import renderer from 'react-test-renderer';

import {createStore} from 'redux';
import {NavigationActions} from 'react-navigation';
import {shallow} from 'enzyme';

// constant
const store = createStore(() => ({}));

describe('List', () => {
  let connectedComponent;
  let wrapper;
  let instance;

  beforeEach(() => {
    connectedComponent = <ConnectedList store={store}/>;

    const connectedWrapper = shallow(connectedComponent);
    wrapper = connectedWrapper.find('List').shallow();
    instance = wrapper.instance();
  });

  it('renders correctly', () => {
    const tree = renderer.create(connectedComponent);
    expect(tree).toMatchSnapshot();
  });

  it('onPressProductDetailButton01 should call navigateToDetail', () => {
    // arrange
    wrapper.setProps({
      navigateToDetail: jest.fn()
    });
    instance = wrapper.instance();

    // act
    instance.onPressProductDetailButton01();

    // assert
    const params = {
      title: 'Car',
      description: 'Vehicle'
    };
    expect(instance.props.navigateToDetail).toHaveBeenCalledWith(params);
  });

  it('onPressProductDetailButton02 should call navigateToDetail', () => {
    // arrange
    wrapper.setProps({
      navigateToDetail: jest.fn()
    });
    instance = wrapper.instance();

    // act
    instance.onPressProductDetailButton02();

    // assert
    const params = {
      title: 'MHW',
      description: 'Game'
    };
    expect(instance.props.navigateToDetail).toHaveBeenCalledWith(params);
  });

  it('onPressProductDetailButton03 should call navigateToDetail', () => {
    // arrange
    wrapper.setProps({
      navigateToDetail: jest.fn()
    });
    instance = wrapper.instance();

    // act
    instance.onPressProductDetailButton03();

    // assert
    const params = {
      title: 'Ice Cream',
      description: 'Dessert'
    };
    expect(instance.props.navigateToDetail).toHaveBeenCalledWith(params);
  });

  it('onPressGoBackButton should call navigateBack', () => {
    // arrange
    wrapper.setProps({
      navigateBack: jest.fn()
    });
    instance = wrapper.instance();

    // act
    instance.onPressGoBackButton();

    // assert
    expect(instance.props.navigateBack).toHaveBeenCalled();
  });

  it('navigateToDetail should call dispatch with params', () => {
    // arrange
    const dispatch = jest.fn();
    const props = mapDispatchToProps(dispatch);
    const params = {
      title: 'Car',
      description: 'Vehicle'
    };

    // act
    props.navigateToDetail(params);

    // assert
    const dispatchParams = NavigationActions.navigate({routeName: 'Detail', params});
    expect(dispatch).toHaveBeenCalledWith(dispatchParams);
  });

  it('navigateBack should call dispatch with params', () => {
    // arrange
    const dispatch = jest.fn();
    const props = mapDispatchToProps(dispatch);
    
    // act
    props.navigateBack();

    // assert
    const dispatchParams = NavigationActions.back();
    expect(dispatch).toHaveBeenCalledWith(dispatchParams);
  });
});
