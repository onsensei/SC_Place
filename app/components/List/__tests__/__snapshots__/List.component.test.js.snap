// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }
  >
    Welcome to List page
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
    Product List
  </Text>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={undefined}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Product 01 &gt;
      </Text>
    </View>
  </View>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={undefined}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Product 02 &gt;
      </Text>
    </View>
  </View>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={undefined}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Product 03 &gt;
      </Text>
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
    Navigate
  </Text>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={undefined}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        &lt; Go Back
      </Text>
    </View>
  </View>
</View>
`;
